source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -23,28 +23,67 @@ RoleBasedAccessControlFilter::generateStats(const std::string& prefix, Stats::Sc
 }
 
 const Filters::Common::RBAC::RoleBasedAccessControlEngine&
-RoleBasedAccessControlFilterConfig::engine(const Router::RouteConstSharedPtr route) const {
+RoleBasedAccessControlFilterConfig::engine(const Router::RouteConstSharedPtr route,
+                                           EnforcementMode mode) const {
   if (!route || !route->routeEntry()) {
-    return engine_;
+    return engine(mode);
   }
 
   const std::string& name = HttpFilterNames::get().RBAC;
   const auto* entry = route->routeEntry();
 
   const auto* route_local =
-      entry->perFilterConfigTyped<Filters::Common::RBAC::RoleBasedAccessControlEngine>(name)
+      entry->perFilterConfigTyped<RoleBasedAccessControlRouteSpecificFilterConfig>(name)
           ?: entry->virtualHost()
-                 .perFilterConfigTyped<Filters::Common::RBAC::RoleBasedAccessControlEngine>(name);
+                 .perFilterConfigTyped<RoleBasedAccessControlRouteSpecificFilterConfig>(name);
 
-  return route_local ? *route_local : engine_;
+  if (route_local) {
+    return route_local->engine(mode);
+  }
+
+  return engine(mode);
 }
 
+bool RoleBasedAccessControlFilterConfig::isEnabled(const Router::RouteConstSharedPtr route,
+                                                   EnforcementMode mode) const {
+  if (!route || !route->routeEntry()) {
```

Comment:
can you extract this part `getRouteLocalConfig` to another function? It is similar with `engine` above.

Reply:
not necessary now as after getting rid of this function.